src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -433,15 +588,26 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost R, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              sandAttack(thisMove)
             }
           }
           move "Flame Drum", {
             text "80 damage. If Magby isn’t anywhere under Magmortar, discard 2 Energy cards from your hand. (If you can’t discard 2 Energy cards from your hand, this attack does nothing.)"
             energyCost R, R, C
-            attackRequirement {}
+            attackRequirement {
+              def condition1 = (self.getPokemonCards().any{it.name.contains("Magby")})
+              def condition2 = (my.hand.filterByType(ENERGY).size() >= 2)
+              assert (condition1 || condition2) : "Magby isn't anywhere under Magmortar, and you can't discard 2 Energy cards from your hand"
+            }
             onAttack {
-              damage 0
+              if (!self.getPokemonCards().any{it.name.contains("Magby")}) {
```

Comment:
If it is already an attackRequirement, I don't think there is any reason to duplicate the condition in an if statement.

Reply:
Ah, ok hadn't seen this comment. Yeah, the check is fine since there are two ways this can be solved. With the change of where damage is declared it should be working properly now, and the magby check here is for doing the discard or not :P